x-pack/test/functional/apps/maps/import_geojson/file_indexing_panel.js

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import path from 'path';
+import uuid from 'uuid/v4';
+
+export default function ({ getService, getPageObjects }) {
+  const PageObjects = getPageObjects(['maps', 'common']);
+  const testSubjects = getService('testSubjects');
+  const log = getService('log');
+
+  const IMPORT_FILE_PREVIEW_NAME = 'Import File';
+  const FILE_LOAD_DIR = 'test_upload_files';
+  const DEFAULT_LOAD_POINT_FILE_NAME = 'point.json';
+  let indexName;
+
+  const indexPoint = async () => await loadFileAndIndex(DEFAULT_LOAD_POINT_FILE_NAME);
+  const indexFileByName = async fileName => await loadFileAndIndex(fileName);
+
+  async function loadFileAndIndex(loadFileName) {
+    await PageObjects.maps.clickAddLayer();
+    // Verify panel page element is open
+    const panelOpen = await PageObjects.maps.isLayerAddPanelOpen();
+    expect(panelOpen).to.be(true);
+    // Select upload source
+    await PageObjects.maps.selectGeoJsonUploadSource();
+    // Upload file
+    log.debug(`Uploading ${loadFileName} for indexing`);
+    await PageObjects.maps.uploadJsonFileForIndexing(
+      path.join(__dirname, FILE_LOAD_DIR, loadFileName)
+    );
+    await PageObjects.maps.waitForLayersToLoad();
+    // Check file upload in TOC
+    const layerLoadedInToc = await PageObjects.maps
+      .doesLayerExist(IMPORT_FILE_PREVIEW_NAME);
+    expect(layerLoadedInToc).to.be(true);
+    // Check file is loaded in file picker
+    const filePickerLoadedFile = await PageObjects.maps
+      .hasFilePickerLoadedFile(loadFileName);
+    expect(filePickerLoadedFile).to.be(true);
+    // Set index name to something random & unrepeating
+    indexName = uuid();
```

Comment:
Instead of using a shared variable. How about returning indexName from `loadFileAndIndex`.